packages/samples/headless-react/src/components/triggers/execute-trigger.class.tsx

```diff
@@ -0,0 +1,45 @@
+import {Component, ContextType} from 'react';
+import {
+  buildExecuteTrigger,
+  ExecuteTrigger as HeadlessExecuteTrigger,
+  ExecuteTriggerState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class ExecuteTrigger extends Component<{}, ExecuteTriggerState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessExecuteTrigger;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildExecuteTrigger(this.context.engine!);
+    this.updateState();
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
```

Comment:
I think the sample would be clearer if we had some simple functions and called them inside a switch or if/else chain. I don't think showing a `render` function is the clearest.

If someone has a classic use-case for the trigger, we can base the sample around it. I will use a generic example to show what I mean:

```
export class ExecuteTrigger {
  componentDidMount() {
      this.controller = buildExecuteTrigger(this.context.engine!);
      this.unsubscribe = this.controller.subscribe(() => this.executeFunction());
  }

  executeFunction() {
    const {name, params} = this.controller.state;
    
    if (name === 'logCatName') {
        logCatName(...params)
    }
    
    ...  
  }
}

function logCatName(name: string) {
    console.log(`cat: ${name}`);
}

function logDogName(name: string) {
    console.log(`dog: ${name}`);
}

```

Reply:
Going further, I don't think it should even be a react component with `componentDidMount` lifecycle, but just a simple class.

Same for the "fn" example below, it does not really need to be a react component per say. Just a function will do.

I think this is what we do the for URL manager ? 🤔